Orange/widgets/visualize/owfreeviz.py

```diff
@@ -810,8 +810,9 @@ def reset_graph_data(self, *_):
             self._update_graph()
 
     def _update_graph(self, reset_view=True, **_):
+        if self.graph.data is None:
```

Comment:
`_update_graph` is called from `reset_graph_data` and `update_density`. The former already checks that `graph.data` is not `None`. You could add this check to the latter as well instead of having it in `_update_graph`, and just move the assert in `_update_graph` to the top.

Reply:
Done.